chain/chain/src/chain.rs

```diff
@@ -703,6 +703,7 @@ impl Chain {
             if chunk_header.height_created() == genesis_block.header().height() {
                 // Special case: genesis chunks can be in non-genesis blocks and don't have a signature
                 // We must verify that content matches and signature is empty.
+                // TODO: this code will not work when genesis block has different number of chunks as the current block
```

Comment:
could we create an issue for this?

Reply:
Yes, I probably won't fix this before the testnet launch though. This won't affect the simple nightshade launch because chunks get produced between the genesis block and when sharding changes

Sorry that I always forget to create issues for TODOs. Sometimes I just added TODOs as I write code and I wasn't sure if they are actually real TODOs and forget about them later .

Reply:
https://github.com/near/nearcore/issues/4908